Description: Enable gcc 4.7 threading detection 
 Patch taken from upstream change set 76133.
Author: Ai Azuma
Bug: 654425
Forwarded: https://svn.boost.org/trac/boost/ticket/6165

--- a/boost/config/stdlib/libstdcpp3.hpp
+++ b/boost/config/stdlib/libstdcpp3.hpp
@@ -34,5 +34,6 @@
 #ifdef __GLIBCXX__ // gcc 3.4 and greater:
 #  if defined(_GLIBCXX_HAVE_GTHR_DEFAULT) \
-        || defined(_GLIBCXX__PTHREADS)
+        || defined(_GLIBCXX__PTHREADS) \
+        || defined(_GLIBCXX_HAS_GTHREADS)
       //
       // If the std lib has thread support turned on, then turn it on in Boost
